src/stationManager.js

```diff
@@ -0,0 +1,115 @@
+import Station from './object/station.js';
+import stationStorage from './storage/stationStorage.js';
+import {STATION_MANAGE_TEMPLATE} from './view/stationTemplate.js'
+
+const MIN_STATION_LENGTH_LIMIT = 2;
+const EMPTY = 0;
+
+const nameDupilcateCheck = (stationName) => {
+  let stationNameArray = stationStorage().getStationNameArray();
+  
+  if(stationNameArray.includes(stationName)) {
+    alert('중복된 이름이 있습니다.');
+    return false;
+  }
+  return true;
+}
+
+const nameLengthCheck = (stationName) => {
+  if(stationName.length < MIN_STATION_LENGTH_LIMIT) {
+    alert('2글자 이상 입력해주세요.');
+    return false;
+  }
+  return true;
+}
+
+const addStationToStorage = (stationName, stationTableBodyTag) => {
+  let newStation = new Station(stationName);
+  let stations = stationStorage().getStation();
+  removeTableBodyRow(stationTableBodyTag, stations);
+
+  stations.push(newStation);
+  stationStorage().setStation(stations);
+  fillStationTableBody(stationTableBodyTag);
+}
+
+const removeTableBodyRow = (stationTableBodyTag, stations) => {
+  for(let i=0;i<stations.length;i++) {
+    stationTableBodyTag.deleteRow(0);
+  }
+}
+
+const fillStationTableBody = (stationTableBodyTag) => {
+  let stations = stationStorage().getStation();
+
+  stations.forEach((station) => {
+    let tbodyRow = stationTableBodyTag.insertRow(stationTableBodyTag.rows.length);
+    let firstCell = tbodyRow.insertCell(0);
+    let secondCell = tbodyRow.insertCell(1);
+    
+    firstCell.innerHTML = station.name;
+    secondCell.innerHTML = '<button class="station-delete-button" data-name=' + station.name + '>삭제</button>';
+  });
```

Comment:
이렇게 `forEach` 로 반복문을 돌면서 `innerHTML` 를 수행하면 리렌더링이 발생한다고 합니다!

그래서 한번에 업데이트 해주는게 좋을 것 같아요. 아래처럼요.

```suggestion
const fillStationTableBody = (stationTableBodyTag) => {
  let stations = stationStorage().getStation();
  let htmlString = "";

  stations.forEach((station) => {
    let tbodyRow = stationTableBodyTag.insertRow(stationTableBodyTag.rows.length);
    let firstCell = tbodyRow.insertCell(0);
    let secondCell = tbodyRow.insertCell(1);
    
    firstCell.innerHTML = station.name;
    htmlString +=  '<button class="station-delete-button" data-name=' + station.name + '>삭제</button>';
  });
  secondCell.innerHTML = htmlString;
```

이런식으로 하나 변수 `htmlString` 으로 하나 빼서 한번에 `innerHTML` 로 접근하면 리렌더링을 최소화할 수 있음요!